sklearn/preprocessing/_function_transformer.py

```diff
@@ -72,18 +100,52 @@ class FunctionTransformer(BaseEstimator, TransformerMixin):
         Dictionary of additional keyword arguments to pass to inverse_func.
 
     """
-    def __init__(self, func=None, inverse_func=None, validate=True,
-                 accept_sparse=False, pass_y='deprecated', check_inverse=True,
-                 kw_args=None, inv_kw_args=None):
+    def __init__(self, func=None, inverse_func=None, validate=None,
+                 accept_sparse=False, force_all_finite=True,
+                 pass_y='deprecated', check_inverse=True, kw_args=None,
+                 inv_kw_args=None):
         self.func = func
         self.inverse_func = inverse_func
         self.validate = validate
         self.accept_sparse = accept_sparse
+        self.force_all_finite = force_all_finite
         self.pass_y = pass_y
         self.check_inverse = check_inverse
         self.kw_args = kw_args
         self.inv_kw_args = inv_kw_args
 
+    def _check_input(self, X):
+        # FIXME: Future warning to be removed in 0.22
+        if self.validate is None:
+            self._validate = True
+            warnings.warn("The default validate=True will be replaced by "
+                          "validate='array-or-frame' in 0.22.", FutureWarning)
```

Comment:
Should we let this depend on the type of X ? Eg only do this is if the input is actually a dataframe? (related to my other question whether the behaviour for arrays will change with regard to NaNs)

Reply:
We can only do it for DataFrame. The behavior for array will not change: we will enforce finiteness by default (no NaN and infinite) which the equivalent of validate=True.